dts/bindings/pinctrl/microchip,xec-pinctrl.yaml

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2020 Linaro Limited
+# Copyright (c) 2021 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Microchip XEC Pin controller Node
+    Based on pincfg-node.yaml binding.
+    The MCHP XEC pin controller is a singleton node responsible for controlling
+    pin function selection and pin properties. For example, you can use this
+    node to select peripheral pin functions. For example, I2C
+
+    The node has the 'pinctrl' node label set in your SoC's devicetree,
+    so you can modify it like this:
+
+      &pinctrl {
+              /* your modifications go here */
+      };
+
+    All device pin configurations should be placed in child nodes of the
+    'pinctrl' node, as shown in this example:
+
+    A group can also specify shared pin properties common to all the specified
+    pins, such as the 'bias-pull-up' property in group 2. Here is a list of
+    supported standard pin properties:
+
+    - bias-disable: Disable pull-up/down (default behavior, not required).
+    - bias-pull-down: Enable pull-down resistor.
+    - bias-pull-up: Enable pull-up resistor.
+    - drive-push-pull: Output driver is push-pull (default, not required).
+    - drive-open-drain: Output driver is open-drain.
+    - output-high: Set output state high when pin configured.
+    - output-low: Set output state low when pin configured.
+
+    Custom pin properties for drive strength and slew rate are available:
+    - drive-strength
+    - slew-rate
+
+    Driver strength and slew rate hardware defaults vary by SoC and pin.
+    The "hw-default" setting tell the pinctrl code to not touch the slew
+    rate or drive strength settings for the respective pin.
+
+    An example for MEC172x family, include the chip level pinctrl
+    DTSI file in the board level DTS:
+
+      #include <microchip/mec172x/mec172xnsz-pinctrl.dtsi>
+
+      We want to use the shared SPI port of the MEC172x QMSPI controller
+      and want the chip select 0 to be open-drain.
+
+      To change a pin's pinctrl default properties add a reference to the
+      pin in the board's DTS file and set the properties.
+
+      &spi0 {
+        pinctrl-0 = < &pinctrl_qint_cs0
+                      &pinctrl_qint_clk
+                      &pinctrl_qint_io0
+                      &pinctrl_qint_io1
+                      &pinctrl_qint_wpn >;
```

Comment:
Please explain the naming convention:

![image](https://user-images.githubusercontent.com/25011557/152314401-ff6d618b-385a-4ed3-b602-a116806b7de8.png)

By looking at datasheet I could understand that `<signal>`, `<signal>_alt` and `<signal>_alt2` are expected node names (no need to specify the `gpio_xxx`), but I couldn't get that from the current binding docs. Since pinctrl DT part has many vendors-specific bits, documentation needs to be clear.

Reply:
This table is just listing those functions that are available on multiple pins to ease board layout.

Please see proposed changes in this [comment ](https://github.com/zephyrproject-rtos/zephyr/pull/41888#discussion_r798974307)and confirm.

Reply:
Naming convention updated as per review comments.